packages/azure-node/src/generators/app-insights-deployment/generator.spec.ts

```diff
@@ -0,0 +1,78 @@
+import { Tree, updateJson } from '@nrwl/devkit';
+import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
+import { applicationGenerator } from '@nrwl/next';
+
+import generator from './generator';
+import { AppInsightsDeploymentGeneratorSchema } from './schema';
+
+describe('app-insights-deployment generator', () => {
+    let appTree: Tree;
+    const options: AppInsightsDeploymentGeneratorSchema = {
+        project: 'next-app',
+        applicationinsightsConnectionString:
+            'APPLICATIONINSIGHTS_CONNECTIONS_STRING',
+    };
+
+    beforeEach(async () => {
+        appTree = createTreeWithEmptyWorkspace();
+        await applicationGenerator(appTree, {
+            name: 'next-app',
+            style: 'css',
+        });
+        updateJson(appTree, 'nx.json', nxJson => ({
+            ...nxJson,
+            stacks: {
+                business: {
+                    company: 'Amido',
+                    domain: 'stacks',
+                    component: 'nx',
+                },
+                domain: {
+                    internal: 'test.com',
+                    external: 'test.dev',
+                },
+                cloud: {
+                    region: 'euw',
+                    platform: 'azure',
+                },
+                pipeline: 'azdo',
+                terraform: {
+                    group: 'terraform-group',
+                    storage: 'terraform-storage',
+                    container: 'terraform-container',
+                },
+                vcs: {
+                    type: 'github',
+                    url: 'remote.git',
+                },
+            },
+        }));
```

Comment:
I see that this is a common prerequisite for the E2E tests. Would it make sense to have a common method, i.e `setStacksJson(appTree);`. That way, if any changes are made to the core nx.json values it would be reflected across all unit tests and help to catch issues in plugins we did not intend to impact?

Reply:
As discussed we are doing this in a later ticket :)